src/main/java/vazkii/quark/world/world/SpeleothemGenerator.java

```diff
@@ -42,11 +42,12 @@ public void generate(Random random, int chunkX, int chunkZ, World world, IChunkG
 			innerSpread = Speleothems.netherClusterCount;
 		}
 		
-		for(int i = 0; i < tries; i++) {
-			BlockPos target = new BlockPos(x + random.nextInt(spread), random.nextInt(upperBound) + offset, z + random.nextInt(spread));
-			if(placeSpeleothemCluster(random, world, target, innerSpread, innerTries))
-				i++;
-		}
+		for(int i = 0; i < tries; i++)
+			if(upperBound > 0) {
```

Comment:
Should check this before the loop, no point checking it once per try when it's constant

Reply:
Yeah, I realized that after committing. I am going to change.